### PR TITLE
#39 make dependent database operations atomic using transactions

### PR DESCRIPTION
**Pull request by wqking**

### CODE REVIEW

**Comment by meyer9** [comment]
Can you rebase this with master?

**Comment by wqking** [comment]
My bad, wrong target branch. Fixed.

**Comment by meyer9** [comment]
Actually, the branch was correct. I merged testnet into master earlier. It should be based on master. Just run git rebase master.

**Comment by wqking** [comment]
Done

File: beacon/state.go [hunk 1]
```diff
@@ -95,21 +95,23 @@ func (b *Blockchain) ProcessBlock(block *primitives.Block, checkTime bool, verif
...
-	// TODO: these two database operations should be in a single transaction
-
 	databaseTipUpdateStart := time.Now()
 
-	// set the block node in the database
-	err = b.db.SetBlockNode(blockNodeToDisk(*node))
-	if err != nil {
-		return err
-	}
+	b.db.TransactionalUpdate(func(transaction interface{}) error {
```
Comment (meyer9) [inline beacon/state.go]: handle error here

**Comment by meyer9** [comment]
LGTM! There are other places where transactional database modifications are needed, but we can add those in over time.

**Pull request merged**